src/Path.hs

```diff
@@ -370,9 +381,12 @@ filename (Path l) =
 --
 -- The following properties hold:
 --
+-- @dirname $(mkRelDir ".") == $(mkRelDir ".")@
+--
 -- @dirname (p \<\/> a) == dirname a@
 --
 dirname :: Path b Dir -> Path Rel Dir
+dirname (Path []) = Path []
```

Comment:
We've stablished that `dirname` is broken: https://github.com/chrisdone/path/issues/18 I don't think we want to make it broken-er. 🙈